apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -676,15 +676,136 @@ function openLinksInParentFrame( calypsoPort ) {
+			const hyperlink = document.querySelector( '.wp-block-query__create-new-link a' );
+			if ( hyperlink ) {
+				hyperlink.href = createNewPostUrl;
+				hyperlink.target = '_top';
+			}
+		} );
+	};
+	const createNewPostLinkObserver = new window.MutationObserver( tryToReplaceCreateNewPostLink );
+
+	// Manage reusable blocks link in the global block inserter's Reusable tab
+	// Post editor only
+	const inserterManageReusableBlocksObserver = new window.MutationObserver( ( mutations ) => {
+		const node = mutations[ 0 ].target;
+		if ( node.attributes.getNamedItem( 'aria-selected' )?.nodeValue === 'true' ) {
+			const hyperlink = document.querySelector( 'a.block-editor-inserter__manage-reusable-blocks' );
+			if ( hyperlink ) {
+				hyperlink.href = manageReusableBlocksUrl;
+				hyperlink.target = '_top';
+			}
+		}
+	} );
+
+	const shouldReplaceCreateNewPostLinksFor = ( node ) =>
+		createNewPostUrl &&
+		( node.classList.contains( 'interface-interface-skeleton__sidebar' ) || // Site editor
+			node.classList.contains( 'edit-post-sidebar' ) ); // Post editor
+
+	const shouldReplaceManageReusableBlockLinksFor = ( node ) =>
+		manageReusableBlocksUrl &&
+		node.classList.contains( 'interface-interface-skeleton__secondary-sidebar' );
+
+	// This observer functions as a "parent" observer, which connects and disconnects
+	// "child" observers as the relevant sidebar settings appear and disappear in the DOM.
+	const sidebarsObserver = new window.MutationObserver( ( mutations ) => {
+		for ( const record of mutations ) {
+			for ( const node of record.addedNodes ) {
+				// Block settings sidebar for Query block.
+				if ( shouldReplaceCreateNewPostLinksFor( node ) ) {
+					const componentsPanel = node.querySelector(
+						'.interface-interface-skeleton__sidebar .components-panel, .edit-post-sidebar .components-panel'
+					);
+					createNewPostLinkObserver.observe( componentsPanel, {
+						childList: true,
+						subtree: true,
+					} );
+					// If a Query block is selected, then the sidebar will
+					// directly open on the block settings tab
+					tryToReplaceCreateNewPostLink();
+				} else if (
+					// Block inserter sidebar, Reusable tab
+					shouldReplaceManageReusableBlockLinksFor( node )
+				) {
+					const resuableTab = node.querySelector(
+						'.components-tab-panel__tabs-item[id*="reusable"]'
+					);
+					if ( resuableTab ) {
+						inserterManageReusableBlocksObserver.observe( resuableTab, {
+							attributeFilter: [ 'aria-selected' ],
+						} );
+					}
+				}
+			}
+
+			for ( const node of record.removedNodes ) {
+				if ( shouldReplaceCreateNewPostLinksFor( node ) ) {
+					createNewPostLinkObserver.disconnect();
+				} else if ( shouldReplaceManageReusableBlockLinksFor( node ) ) {
+					inserterManageReusableBlocksObserver.disconnect();
+				}
+			}
+		}
+	} );
+	// Site editor
+	sidebarsObserver.observe( document.querySelector( '.interface-interface-skeleton__body' ), {
+		childList: true,
+	} );
+	// Post editor
```

Comment:
I think this is for the block settings sidebar in both the post and site editors?